pkg/restore/restore.go

```diff
@@ -1098,19 +1115,30 @@ func (ctx *restoreContext) restoreItem(obj *unstructured.Unstructured, groupReso
 		}
 	}
 
-	// Clear out non-core metadata fields and status.
-	if obj, err = resetMetadataAndStatus(obj); err != nil {
+	// Clear out non-core metadata fields.
+	if obj, err = resetMetadata(obj); err != nil {
 		errs.Add(namespace, err)
 		return warnings, errs
 	}
 
+	shouldRestoreStatus := ctx.resourceStatusIncludesExcludes.ShouldInclude(groupResource.String())
+
+	ctx.log.Infof("restore status includes excludes: %+v", ctx.resourceStatusIncludesExcludes)
+	// Clear out status.
+	if !shouldRestoreStatus {
+		ctx.log.Infof("Resetting status for obj %s/%s", obj.GetKind(), obj.GetName())
+		if obj, err = resetStatus(obj); err != nil {
```

Comment:
It's better put this reset further below. As you can see, when processing RestoreItemAction, the obj may be replaced with the return value of the item action, so the reset to obj here may be overwritten. Maybe we can put it right above resourceClient.Create(obj)?

Reply:
I was trying to change as little as possible, but sure, we can do that :)